api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -430,6 +435,178 @@ public void saveProgram_shouldCreateProgramWorkflows() {
 		}
 		TestUtil.assertCollectionContentsEquals(Arrays.asList("SINGLE", "MARRIED"), names);
 	}
+
+	/**
+	 * Tests if saveProgram throws API Exception with program, workflows and states having null concept 
+	 *
+	 * @see ProgramWorkflowService#saveProgram(Program)
+	 */
+	@Test
+	public void saveProgram_shouldThrowAPIExceptionGivenNullConcept() {
+		Program program = new Program();
+		program.setName("TEST PROGRAM");
+		program.setDescription("TEST PROGRAM DESCRIPTION");
+		Assertions.assertThrows(APIException.class, ()->{
+			pwsi.saveProgram(program);
+		});
+
+		program.setConcept(cs.getConcept(3));
```

Comment:
doesnt the concept seem not to be set here yet it seems like you need to have it as a null

Reply:
I had it null before. I'm setting concept for program here and then checked for the case when the concept for workflow is null.